knimin/tests/test_ag_third_party.py

```diff
@@ -9,19 +9,25 @@
 from knimin.tests.tornado_test_base import TestHandlerBase
 from knimin import db
 from knimin.handlers.ag_third_party import ThirdPartyData, NewThirdParty
+from functools import partial
 
 
 class AGThirdPartyHandler(TestHandlerBase):
     ext_survey_fp = join(dirname(realpath(__file__)), 'data',
                          'external_survey_data.csv')
-
-    def setUp(self):
-        # Make sure vioscreen survey exists in DB
-        try:
-            db.add_external_survey('Vioscreen', 'FFQ', 'http://vioscreen.com')
-        except ValueError:
-            pass
-        super(AGThirdPartyHandler, self).setUp()
+    data_vioscreen = {'name': 'Vioscreen',
+                      'description': 'FFQ',
+                      'url': 'http://vioscreen.com'}
+    _clean_up_funcs = []
+
+    def tearDown(self):
+        for f in self._clean_up_funcs:
+            try:
+                f()
+            except:
+                print("clean-up failed for. Downstream tests might be affected"
```

Comment:
It means that something has not been deleted in the DB which was inserted previously for executing unit tests.
I think also specifying what failed will be some overhead development and goes into the direction of the already existing teadDown() method.